scripts/weekly-summary

```diff
@@ -0,0 +1,374 @@
+import os
+import sys
+from subprocess import check_call
+import tempfile
+from urllib.request import Request, urlopen
+import zipfile
+
+import pycurl
+
+URL = "https://api.github.com/repos/rhinstaller/kickstart-tests/actions/artifacts?per_page=100"
+
+# TODO - only works after we have a week of json summary runs
+#ARTIFACTS = ["summary-daily-iso", "summary-rhel9", "summary-rhel8"]
+
+# For now use the full log artifacts
+ARTIFACTS = ["logs-daily-iso", "logs-rhel9", "logs-rhel8"]
+
+
+# We should ignore SIGPIPE when using pycurl.NOSIGNAL - see
+# the libcurl tutorial for more info.
+try:
+    import signal
+    from signal import SIGPIPE, SIG_IGN
+except ImportError:
+    pass
+else:
+    signal.signal(SIGPIPE, SIG_IGN)
+
+
+def get_artifacts(token, start, end):
+    """
+    get_artifacts retrieves a list of artifact from the selected date range
+
+    It returns a list of artifact dicts like this:
+
+    {
+      "id": 187278866,
+      "node_id": "MDg6QXJ0aWZhY3QxODcyNzg4NjY=",
+      "name": "logs-daily-iso",
+      "size_in_bytes": 1047408950,
+      "url": "https://api.github.com/repos/rhinstaller/kickstart-tests/actions/artifacts/187278866",
+      "archive_download_url": "https://api.github.com/repos/rhinstaller/kickstart-tests/actions/artifacts/187278866/zip",
+      "expired": false,
+      "created_at": "2022-03-17T03:30:12Z",
+      "updated_at": "2022-03-17T03:30:13Z",
+      "expires_at": "2022-03-25T03:29:06Z"
+    },
+    """
+    req = Request(URL)
+    req.add_header("Accept", "application/vnd.github.v3+json")
+    req.add_header("Authorization", f"token {token}")
+    with urlopen(req) as r:
+        data = json.load(r)
+
+    # Filter out the artifacts within the date range and names
+    artifacts = []
+    for a in data["artifacts"]:
+        if a["name"] not in ARTIFACTS:
+            continue
+        updated_at = datetime.fromisoformat(a["updated_at"][:-1])
+        if start <= updated_at <= end:
+            artifacts.append(a)
+
+    return sorted(artifacts, key=lambda x: x["updated_at"])
+
+
+def run_curl(token, url, filename):
+    """
+    run_curl downloads an artifact file
+    """
+    with open(filename, "wb") as f:
+        c = pycurl.Curl()
+        headers = [
+           "Accept: application/vnd.github.v3+json",
+           f"Authorization: token {token}",
+        ]
+
+        options = {
+            pycurl.FOLLOWLOCATION:  1,
+            pycurl.MAXREDIRS:       5,
+            pycurl.CONNECTTIMEOUT:  30,
+            pycurl.TIMEOUT:         300,
+            pycurl.NOSIGNAL:        1,
+            pycurl.URL:             url,
+            pycurl.HTTPHEADER:      headers,
+            pycurl.WRITEDATA:       f
+        }
+        for k, v in options.items():
+            c.setopt(k, v)
+
+        try:
+            c.perform()
+            ok = (True, None)
+        except Exception as e:
+            print(f"ERROR: {e}", file=sys.stderr)
+            sys.stderr.flush()
+            ok = (False, e)
```

Comment:
I think an exception is more appropriate here but I don't mind.

Reply:
That's on purpose. I want to log problems, but not stop the process. That's also why it doesn't check the return code.